lib/python/dtype.cpp

```diff
@@ -36,15 +36,15 @@ constexpr bool operator==(const char a, const DTypeKind b) {
   return a == static_cast<char>(b);
 }
 
-enum class DTypeSize : scipp::index {
+enum class DTypeSize : ssize_t {
```

Comment:
Digging around the git blame this used to be `ssize_t` but changed to `scipp::index` in https://github.com/scipp/scipp/pull/2313 to make windows happy. I'm not sure if this is still required.

This seems to fix the issue with numpy2.0 and I'm not a 100% sure why! (there are other failures now) I just tried to sync types with pybind: 
https://github.com/pybind/pybind11/blob/750257797ca5eda6f1561f6c3f7147bcdd92cf52/include/pybind11/numpy.h#L563